controlplane/pkg/plugins/utils.go

```diff
@@ -38,6 +39,32 @@ func StartPlugin(name, registry string, services map[plugins.PluginCapability]in
 		return err
 	}
 
+	go func() {
+		registered := true
+		for ; ; time.Sleep(5 * time.Second) {
+			resp, err := http.Get("http://localhost:5555/liveness")
+			if err != nil {
+				logrus.Errorf("failed to get NSMD liveness: %v", err)
+			}
+
+			nsmdAlive := err == nil && resp.StatusCode >= 200 && resp.StatusCode < 400
+			if nsmdAlive && !registered {
+				if err := registerPlugin(name, endpoint, registry, capabilities); err != nil {
+					logrus.Errorf("failed to re-register plugin '%v' at '%v'", name, endpoint)
+				} else {
+					logrus.Infof("Successfully re-registered plugin '%v' at '%v'", name, endpoint)
+					registered = true
+				}
+			}
+			if !nsmdAlive && registered {
+				logrus.Error("NSMD failure detected, waiting for it to restart...")
+				registered = false
+			}
+			// nsmdAlive && registered: all is well, do nothing
```

Comment:
I think that comments quite obvious, variables names are comprehensive enough

Reply:
- [ ] remove obvious comments